gemini/prompts/prompt_optimizer/vertex_ai_prompt_optimizer_ui.ipynb

```diff
@@ -0,0 +1,907 @@
+        "OPTIMIZATION_METRIC_1_WEIGHT = 0.0  # @param {type:\"number\"}\n",
+        "OPTIMIZATION_METRIC_2 = \"NA\"  # @param [\"NA\", \"bleu\", \"coherence\", \"exact_match\", \"fluency\", \"groundedness\", \"text_quality\", \"verbosity\", \"rouge_1\", \"rouge_2\", \"rouge_l\", \"rouge_l_sum\", \"safety\", \"question_answering_correctness\", \"question_answering_quality\", \"summarization_quality\", \"tool_name_match\", \"tool_parameter_key_match\", \"tool_parameter_kv_match\", \"tool_call_valid\"] {type:\"string\"}\n",
+        "OPTIMIZATION_METRIC_2_WEIGHT = 0.0  # @param {type:\"number\"}\n",
+        "OPTIMIZATION_METRIC_3 = \"NA\"  # @param [\"NA\", \"bleu\", \"coherence\", \"exact_match\", \"fluency\", \"groundedness\", \"text_quality\", \"verbosity\", \"rouge_1\", \"rouge_2\", \"rouge_l\", \"rouge_l_sum\", \"safety\", \"question_answering_correctness\", \"question_answering_quality\", \"summarization_quality\", \"tool_name_match\", \"tool_parameter_key_match\", \"tool_parameter_kv_match\", \"tool_call_valid\"] {type:\"string\"}\n",
+        "OPTIMIZATION_METRIC_3_WEIGHT = 0.0  # @param {type:\"number\"}\n",
+        "METRIC_AGGREGATION_TYPE = \"weighted_sum\"  # @param [\"weighted_sum\", \"weighted_average\"]\n",
+        "# @markdown **Misc Configs**: <br/>\n",
+        "# @markdown * This variable is used for long prompt optimization to not optimize parts of prompt identified by placeholders. It provides a mapping from the placeholder variables to their content. See link for details.\n",
+        "PLACEHOLDER_TO_VALUE = \"{}\"  # @param\n",
+        "# @markdown * This variable determines the format of the output for the target model. See link for details.\n",
+        "RESPONSE_MIME_TYPE = \"application/json\"  # @param [\"text/plain\", \"application/json\"]\n",
+        "# TODO: add the description for the language.\n",
+        "TARGET_LANGUAGE = \"English\"  # @param [\"English\", \"French\", \"German\", \"Hebrew\", \"Hindi\", \"Japanese\", \"Korean\", \"Portuguese\", \"Simplified Chinese\", \"Spanish\", \"Traditional Chinese\"]"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "X7Mgb0EHSSFk"
+      },
+      "source": [
+        "# Step 5: Run Prompt Optimizer"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "Z8NvNLTfxPTf"
+      },
+      "outputs": [],
+      "source": [
+        "timestamp = datetime.datetime.now().strftime(\"%Y-%m-%dT%H:%M:%S\")\n",
+        "display_name = f\"pt_{timestamp}\"\n",
+        "\n",
+        "in_colab_enterprise = \"GOOGLE_CLOUD_PROJECT\" in os.environ\n",
+        "if not in_colab_enterprise:\n",
+        "    gc = authenticate()\n",
```

Comment:
Can you move this to the top (similar to the notebook template)?

Reply:
Our PM wants this notebook behave like a UI with as few steps as possible that's why we didn't make this a separate step.